AmethystAPI/src/amethyst/Memory.h

```diff
@@ -8,6 +8,30 @@
 #include <vector>
 #include <windows.h>
 
+template <typename Ret, typename Type>
+Ret& direct_access(Type* type, size_t offset) {
+    union {
+        size_t raw;
+        Type* source;
+        Ret* target;
+    } u;
+    u.source = type;
+    u.raw += offset;
+    return *u.target;
+}
+
+#define AS_FIELD(type, name, fn) __declspec(property(get = fn, put = set##name)) type name
+#define DEF_FIELD_RW(type, name) __declspec(property(get = get##name, put = set##name)) type name
+
+#define FAKE_FIELD(type, name)       \
+	AS_FIELD(type, name, get##name); \
+	type get##name()
+
+#define BUILD_ACCESS(ptr, type, name, offset)                           \
+	AS_FIELD(type, name, get##name);                                    \
+	type get##name() const { return direct_access<type>(ptr, offset); } \
```

Comment:
just noticed this now... this should return a reference to type, otherwise unnecessary copies are being created. seems extremely bug prone.